pkgs/applications/networking/remote/rustdesk/default.nix

```
@@ -0,0 +1,113 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, makeDesktopItem
+, copyDesktopItems
+, makeWrapper
+, rustPlatform
+, cmake
+, yasm
+, nasm
+, zip
+, pkg-config
+, clang
+, gtk3
+, xdotool
+, libxcb
+, libXfixes
+, alsa-lib
+, pulseaudio
+, libXtst
+, libvpx
+, libyuv
+, libopus
+, libsciter
+, llvmPackages
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "rustdesk";
+  version = "1.1.8";
+
+  src = fetchFromGitHub {
+    owner = pname;
+    repo = pname;
+    rev = "0325500ebf331b66220cec6e9078afb83b0e98a7";
+    sha256 = "sha256-xglyyoiAjJx3y8+A2OYHZffjqjDkcTjIluPA/J42VVg=";
+  };
+
+  cargoSha256 = "sha256-4MQKa54f3X7IHGd29H6RY7v2toeHvTHInIpgXjdotjw=";
+
+  LIBCLANG_PATH="${llvmPackages.libclang.lib}/lib";
+
+  # Change magnus-opus version to upstream so that it does not use
+  # vcpkg for libopus since it does not work.
+  cargoPatches = [
+    ./cargo.patch
+  ];
+
+  # Manually simulate a vcpkg installation so that it can link the libaries
+  # properly.
+  postUnpack = let
+    target_arch = if stdenv.isx86_64 then "x64"
+      else if stdenv.isx86_32 then "x32"
+      else throw "Unsupported architecture";
+
+    target_os = if stdenv.isLinux then "linux"
+      else if stdenv.isDarwin then "osx"
+      else throw "Unsupported system";
+
+    vcpkg_target = "${target_arch}-${target_os}";
+  in ''
+    export VCPKG_ROOT="$TMP/vcpkg";
+
+    mkdir -p $VCPKG_ROOT/installed/${vcpkg_target}/lib
+
+    ln -s ${libvpx.out}/lib/* $VCPKG_ROOT/installed/${vcpkg_target}/lib/
+    ln -s ${libyuv.out}/lib/* $VCPKG_ROOT/installed/${vcpkg_target}/lib/
+  '';
+
+  nativeBuildInputs = [ pkg-config cmake makeWrapper copyDesktopItems yasm nasm clang ];
+  buildInputs = [ alsa-lib pulseaudio libXfixes libxcb xdotool gtk3 libvpx libopus libXtst libyuv ];
+
+  # Checks require an active X display.
+  doCheck = false;
+
+  desktopItem = makeDesktopItem rec {
+    name = pname;
+    exec = pname;
+    icon = pname;
+    desktopName = "RustDesk";
+    comment = "Yet another remote desktop software";
+    genericName = "Remote Desktop";
+    categories = "Network";
+  };
+
+  desktopItems = [ desktopItem ];
```

You could do this without creating the `desktopItem` attribute:
```suggestion
  desktopItems = [ (makeDesktopItem {
    name = "restdesk";
    exec = meta.mainProgram;
    icon = "rustdesk";
    desktopName = "RustDesk";
    comment = "Yet another remote desktop software";
    genericName = "Remote Desktop";
    categories = "Network";
  }) ];
```
I would also advice not to use `pname` but instead just typing it out, just to make it a bit more readable.